ChangeLog.d/fix-redefination_warning_messages_for_GNU_SOURCE.txt

```diff
@@ -0,0 +1,5 @@
+Bugfix
+   * Fix issue of redefinition warning messages for _GNU_SOURCE in 
+     entropy_poll.c and sha_256.c. There was a build warning during 
+     building for linux platform.
+     Resolves #9026
```

Comment:
Spaces at EOL, and file does not end with a newline - this is causing the CI fail

Reply:
Cherry-picked necessary commit. Thanks for the suggestion.